### ISSUE TITLE
Could not determine closest housenumber on an osm interpolation line

### ISSUE THREAD

**Issue by cturanhc** [osm-search/Nominatim#2630]
Planet Import to my private server, some reverse geocoding requests response 500 HTTP status code and says "Could not determine closest housenumber on an osm interpolation line.". When i check the postgresql logs, there is an error like;

```
nominatim@nominatim ERROR:  line_locate_point: 1st arg isn't a line
```
```
nominatim@nominatim STATEMENT:  SELECT place_id, parent_place_id, 26 as rank_search,  (endnumber - startnumber) * ST_LineLocatePoint(linegeo,ST_SetSRID(ST_Point(23.717132,37.979388),4326)) as fhnr,  startnumber, endnumber, step,  ST_Distance(linegeo,ST_SetSRID(ST_Point(23.717132,37.979388),4326)) as distance FROM location_property_osmline WHERE ST_DWithin(ST_SetSRID(ST_Point(23.717132,37.979388),4326), linegeo, 0.001) and indexed_status = 0 and startnumber is not NULL  ORDER BY distance ASC limit 1;
```

I follow the all instructions from Nominatim 4.0.1 Docs and there is no error on my old Nominatim 3.5.2 version.
**Request to my Nominatim 4.0.1 installation;**
http://x.x.x.x/reverse.php?lat=37.979388&lon=23.717132&addressdetails=1&zoom=18

**Response from my Nominatim 4.0.1 installation;**
```
<error>
<code>500</code>
<message>Could not determine closest housenumber on an osm interpolation line.</message>
</error>
```

**Request to nominatim.openstreetmap.org;**
https://nominatim.openstreetmap.org/reverse.php?lat=37.979388&lon=23.717132&addressdetails=1&zoom=18

**Response from nominatim.openstreetmap.org;**
```
<reversegeocode timestamp="Mon, 14 Mar 22 11:30:04 +0000" attribution="Data © OpenStreetMap contributors, ODbL 1.0. http://www.openstreetmap.org/copyright" querystring="lat=37.979388&lon=23.717132&addressdetails=1&zoom=18">
<result place_id="60843476" osm_type="way" osm_id="582254010" lat="37.97977095" lon="23.71745035" boundingbox="37.97972095,37.97982095,23.71740035,23.71750035" place_rank="30" address_rank="30">3, Πλαταιών, Μεταξουργείο, Rouf Suburb, 3rd District of Athens, Athens, Municipality of Athens, Regional Unit of Central Athens, Attica, 10435, Greece</result>
<addressparts>
<house_number>3</house_number>
<road>Πλαταιών</road>
<neighbourhood>Μεταξουργείο</neighbourhood>
<suburb>Rouf Suburb</suburb>
<city_district>3rd District of Athens</city_district>
<city>Athens</city>
<municipality>Municipality of Athens</municipality>
<county>Regional Unit of Central Athens</county>
<state_district>Attica</state_district>
<state>Attica</state>
<postcode>10435</postcode>
<country>Greece</country>
<country_code>gr</country_code>
</addressparts>
</reversegeocode>
```

- **Versions**
```
Nominatim version: 4.0.1
Postgresql version: 12.9-0ubuntu0.20.04.1
Postgis version: POSTGIS="3.0.0 r17983" [EXTENSION] PGSQL="120" GEOS="3.8.0-CAPI-1.13.1 " PROJ="6.3.1" LIBXML="2.9.4" LIBJSON="0.13.1" LIBPROTOBUF="1.3.3" WAGYU="0.4.3 (Internal)"
OS: Ubuntu 20.04.4
```
- **Server Specs**
```
RAM: 64G
number of CPUs: 16
type and size of disks: 2TB NVMe
bare metal/AWS/other cloud service: Bare Metal
```
- **PostgreSQL Config**
```
max_connections = 400                   # (change requires restart)
shared_buffers = 4GB                    # min 128kB
work_mem = 50MB                         # min 64kB
maintenance_work_mem = 10GB             # min 1MB
autovacuum_work_mem = 2GB               # min 1MB, or -1 to use maintenance_work_mem
wal_level = minimal                     # minimal, replica, or logical
checkpoint_timeout = 60min              # range 30s-1d
max_wal_size = 2GB
min_wal_size = 500MB
checkpoint_completion_target = 0.9      # checkpoint target duration, 0.0 - 1.0
max_wal_senders = 0             # max number of walsender processes
effective_cache_size = 24GB
```

Some other URI with responses same error;
/reverse.php?lat=53.290572&lon=-6.680487&addressdetails=1&zoom=18&format=json
/reverse.php?lat=53.290187&lon=-6.68047&addressdetails=1&zoom=18&format=json
/reverse.php?lat=53.29051&lon=-6.680488&addressdetails=1&zoom=18&format=json
/reverse.php?lat=53.289843&lon=-6.680253&addressdetails=1&zoom=18&format=json
/reverse.php?lat=53.28978&lon=-6.68046&addressdetails=1&zoom=18&format=json
/reverse.php?lat=53.289783&lon=-6.680437&addressdetails=1&zoom=18&format=json

**Comment by cturanhc** [comment]
I dropped the nominatim database from PostgreSQL and reimport whole planet again. Problem still occurs. I don't think this is about corrupted database.

**Comment by lonvia** [comment]
The error looks to be coming from a Nominatim master version not 4.0.1.

**Comment by lonvia** [comment]
That said, I see the issue in the code [here](https://github.com/osm-search/Nominatim/blob/930a5cd12a614afa48b06318d8227e6d11f2ce5b/lib-php/ReverseGeocode.php#L67). Needs special casing for startnumber = endnumber and a test.

**Comment by cturanhc** [comment]
> The error looks to be coming from a Nominatim master version not 4.0.1.

Oh... I'm sorry. You're right it's master version. Not 4.0.1. So how can i help you for tracing the issue?

**Comment by lonvia** [comment]
Fixed by #2633.
